### PR TITLE
Move longitude_scale() into Location class

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

File: ArduSub/commands.cpp [hunk 1]
```diff
@@ -70,8 +70,6 @@ bool Sub::set_home(const Location& loc, bool lock)
 
     // init inav and compass declination
     if (!home_was_set) {
-        // update navigation scalers.  used to offset the shrinking longitude as we go towards the poles
-        scaleLongDown = longitude_scale(loc);
```
Comment (WickedShell) [inline ArduSub/commands.cpp]: Looks to be a behavior change.

Comment (peterbarker) [reply ArduSub/commands.cpp]: Fixed.

**Comment by tridge** [comment]
hmm, where is the cache we had for that value?

**Comment by WickedShell** [comment]
> hmm, where is the cache we had for that value?

To my memory we used to have a cache years ago when we supported the APM2.x boards, but it was only on those builds, and we explicitly recalculated it if you were a faster CPU class anyways, and when we dropped the support for the APM2.x stuff it was removed.

**Pull request merged**